Neos.Flow/Classes/Mvc/ActionResponse.php

```diff
@@ -196,7 +196,7 @@ public function getStatusCode(): int
      */
     public function getContentType(): string
     {
-        return $this->contentType;
+        return $this->contentType ?? '';
```

Comment:
Just an alternative solution that makes the behavior more deterministic IMO.

```suggestion
        if ($this->contentType === null) {
            throw new \RuntimeException('No content type is set, use hasContentType() to determine that case', 1618240401);
        }
        return $this->contentType;
```

Reply:
Not sure if throwing in that case is a good measure. It's pretty drastic, given the user just wanted to read the content-type set on the ActionResponse, no?
I see it like this: If you read the ContentType, which is a string, you should anyway handle an empty string (or we use `non-empty-string` psalm type and check for that in the setter). So at that point, what is the difference between having set an empty string and nothing having been set on the `ActionResponse`? We could also instead initialize `$contentType = '';`.

I would probably still prefer the exception to a nullable return type though

Reply:
>  you should anyway handle an empty string

if you _know_ that you need to handle an empty string, you can do the `hasContentType()` check as well.
The difference is: an exception _forces_ you to handle it in case you were doing something nasty like
```php
$filename = $someSlug . '.' . $this->response->getContentType();
```

> (or we [...] and check for that in the setter)

Yes, why not? :)


I'm fine without the exception obviously and I agree that this is an edge case of an "unexpected" behavior